staging/src/k8s.io/client-go/tools/record/fake.go

```diff
@@ -54,7 +55,12 @@ func (f *FakeRecorder) Eventf(object runtime.Object, eventtype, reason, messageF
 }
 
 func (f *FakeRecorder) AnnotatedEventf(object runtime.Object, annotations map[string]string, eventtype, reason, messageFmt string, args ...interface{}) {
-	f.Eventf(object, eventtype, reason, messageFmt, args...)
+	if f.IncludeAnnotations {
+		args = append(args, fmt.Sprint(annotations))
+		f.Eventf(object, eventtype, reason, messageFmt+" %s", args...)
```

Comment:
The args are meant to displayed based on the message format provided, so if we do that, we will just be adding the annotations as part of the message which shouldn't be what we want.

Instead we could reuse the same string as in `Eventf` and add the annotations at the end of it, after the involvedObject.

Reply:
I'll change the approach to do this 👍🏻